packages/seacas/libraries/ioss/src/exodus/Ioex_BaseDatabaseIO.h

```diff
@@ -90,6 +90,94 @@ namespace Ioex {
     // database supports that type (e.g. return_value & Ioss::FACESET)
     IOSS_NODISCARD unsigned entity_field_support() const override;
 
+    IOSS_NODISCARD std::string get_internal_change_set_name() const override { return m_groupName; }
+
+    /** \brief Checks if a database type supports groups
+     *
+     *  \returns True if successful.
+     */
+    bool supports_group()
+    {
+      IOSS_FUNC_ENTER(m_);
+      return supports_group_nl();
+    }
+
+    /** \brief If a database type supports groups and if the database
+     *         contains groups, open the specified group.
+     *
+     *  If the group_name begins with '/', it specifies the absolute path
+     *  name from the root with '/' separating groups.  Otherwise, the
+     *  group_name specifies a child group of the currently active group.
+     *  If group_name == "/" then the root group is opened.
+     *
+     *  \param[in] group_name The name of the group to open.
+     *  \returns True if successful.
+     */
+    bool open_group(const std::string &group_name)
```

Comment:
Similar.  Is there something that another thread coulf do to this file that requires us to get a lock...

And if so, is there another locked function that calls `open_group` which would require us to provide a no-lock version...

Probably something we need to go through all functions and verify 1) do they need to get a lock. and if so 2) does another locked function call this function (so we need both a lock and no-lock version).

Reply:
I believe `open_group` is called from a locked function but I'd like to go through them with you to confirm which needs to have lock/no-lock versions